### PR TITLE
Some update to the LSTM code.

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: code/lstm.py [hunk 1]
```diff
@@ -617,13 +545,17 @@ def main(job_id, params):
...
         'encoder': ['lstm'],
         'dim-proj': [128],
         'n-words': [10000],
-        'optimizer': ['adadelta'],
-        'activ': ['lambda x: tensor.tanh(x)'],
-        'decay-c': [0.],
-        'use-dropout': [1],
+        'optimizer': [adadelta],  # sgd, adadelta and rmsprop available
+        'activ': [tensor.tanh],  # The activation function from Theano.
+        'decay-c': [0.], #
```
Comment (carriepl) [inline code/lstm.py]: Should a comment be here or was the '#' a typo?

File: code/lstm.py [hunk 2]
```diff
@@ -617,13 +545,17 @@ def main(job_id, params):
...
+        'use-dropout': [1],  # if disable slightly faster, but worst test error.
```
Comment (carriepl) [inline code/lstm.py]: 'disabled' ?

File: code/lstm.py [hunk 3]
```diff
@@ -429,24 +358,23 @@ def pred_error(f_pred, prepare_data, data, iterator, verbose=False):
...
-          n_words=100000,
-          data_sym=False,
-          optimizer='rmsprop',
-          encoder='rconv',
-          saveto='model.npz',
+          dispFreq=100,  # display to stdout the training progress every N updates
+          activ=tensor.tanh,
+          decay_c=0.,  # weight decay for the classifier
+          lrate=0.01,  # learning rate for sgd (not used for adadelta and rmsprop)
+          n_words=100000,  # wocabulary size
```
Comment (carriepl) [inline code/lstm.py]: vocabulary

**Comment by nouiz** [comment]
Can someone review this PR and merge it? I think it do a good code clean up and updates.

File: code/lstm.py [hunk 4]
```diff
@@ -590,40 +553,18 @@ def train(dim_proj=100,
                 valid_err=valid_err, test_err=test_err,
                 history_errs=history_errs, **params)
 
+    print 'The code run for %d epochs, with %f sec/epochs' % (
+        (eidx + 1), 1. * (eidx + 1) / (end_time - start_time))
+    print >> sys.stderr, ('Training took %.1fs minutes' %
```
Comment (carriepl) [inline code/lstm.py]: "minutes" => "seconds"?

File: code/lstm.py [hunk 5]
```diff
@@ -428,26 +377,30 @@ def pred_error(f_pred, prepare_data, data, iterator, verbose=False):
...
+    saveFreq=100000,  # save the parameters after every saveFreq updates
+    maxlen=100,  # longer sequence get ignored
+    batch_size=64,
+    valid_batch_size=64,
+    dataset='imdb',
+
+    # Parameter for extra option
+    noise_std=0.,
+    use_dropout=False,  # if False slightly faster, but worst test error
+                        # TODO: This frequently need a bigger model.
```
Comment (carriepl) [inline code/lstm.py]: No apparent need for the "TODO"

File: code/lstm.py [hunk 6]
```diff
@@ -428,26 +377,30 @@ def pred_error(f_pred, prepare_data, data, iterator, verbose=False):
...
+    dim_proj=128,  # TODO: What is this
+    patience=10,  # number of epoch to wait before early stop if no progress
+    max_epochs=5000,  # The maximum number of epoch to run
+    dispFreq=10,  # display to stdout the training progress every N updates
+    activ=tensor.tanh,  # The activation function from Theano.
+    decay_c=0.,  # weight decay for the classifier applied to the U weights.
+    lrate=0.0001,  # learning rate for sgd (not used for adadelta and rmsprop)
+    n_words=10000,  # vocabulary size
+    optimizer=sgd,  # sgd, adadelta and rmsprop available
+    encoder='lstm',  # TODO: can be removed must be lstm.
```
Comment (carriepl) [inline code/lstm.py]: Can you remove this parameter?

File: code/lstm.py [hunk 7]
```diff
@@ -428,26 +377,30 @@ def pred_error(f_pred, prepare_data, data, iterator, verbose=False):
...
-          noise_std=0.,
-          validFreq=1000,
-          saveFreq=1000,  # save the parameters after every saveFreq updates
-          maxlen=50,
-          batch_size=16,
-          valid_batch_size=16,
-          dataset='sentiment140',
-          use_dropout=False):
+def test_lstm(
+    dim_proj=128,  # TODO: What is this
```
Comment (carriepl) [inline code/lstm.py]: I would guess it's the size of the projection (hidden recurrent representation). We should make sure and put it in the comment.

**Comment by carriepl** [comment]
Ok, this looks good enough for merge. Good work.

**Pull request merged**